### ISSUE TITLE
Error Message

### ISSUE THREAD

**Issue by cliffordsnow** [tyrasd/overpass-turbo#171]
I am getting the following error message: `An error occured during the execution of the overpass query!
Request rejected. (e.g. server not found, request blocked by browser addon, request redirected, internal server errors, etc.) Error-Code: error (0)`

Same error message from chrome and firefox on OS X and Ubuntu 

The query is
```
[out:json][timeout:400];
// fetch area “king county Washington” to search in
{{geocodeArea:king county Washington}}->.searchArea;
// gather results
(
  // query part for: “amenity=restaurant”
  node(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="restaurant"](area.searchArea);
  way(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="restaurant"](area.searchArea);
  relation(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="restaurant"](area.searchArea);
  // query part for: “amenity=cafe”
  node(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="cafe"](area.searchArea);
  way(newer:"2010-01-01T07:00:00Z")["amenity"="cafe"](area.searchArea);
  relation(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="cafe"](area.searchArea);
  // query part for: “amenity=fast_food”
  node(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="fast_food"](area.searchArea);
  way(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="fast_food"](area.searchArea);
  relation(changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z")["amenity"="fast_food"](area.searchArea);
);
// print results
out body;
>;
out skel qt;
```

**Comment by tyrasd** [comment]
Hey. Sorry for the late reply, I've somehow overlooked this ticket.

Confirming. The server responds with a `Failed to load resource: the server responded with a status of 504 (Gateway Timeout)` error after a couple of minues, but doesn't send an `Access-Control-Allow-Origin` HTTP-header with the error message. Therefore the site's javascript code isn't even allowed to access the error code, resulting in the very generic error message above.

//ping @drolbr: This looks like a server config issue, doesn't it?

**Comment by tyrasd** [comment]
@cliffordsnow oh, and by the way: the Overpass API doesn't hold OSM history data for dates before the ODbL license change (which was in September 2012), so `changed:"2010-01-01T07:00:00Z","2011-01-01T06:59:59Z"` probably won't really return anything meaningful, I'm afraid.

**Comment by cliffordsnow** [comment]
Now that you bring up the license change I remember being told that. I'm trying to show a county official how OSM has grown in the county. But I can live with the changes since 2012. Won't be as dramatic without earlier data but sufficient for my purposes. I'll see if the smaller query works.

**Comment by cliffordsnow** [comment]
Changing the date to September 2012 did not make any difference in the error.

**Comment by drolbr** [comment]
> @drolbr: This looks like a server config issue, doesn't it?

Unfortunately, this is a bigger issue. As far as it looks, the software sends the header.

However, the standards aren't that clear. And worse, a lot of even common software like wget doesn't fully adhere to standards. Therefore I would like to ask someone for help. The server has, on an abstract level, six possible outcomes of a request:
a) a full response to a valid query
b) a response with runtime error to a valid query
c) a valid query that is refused due to global load shedding
d) a valid query that is refused due to IP specfic quotas
e) an ill-formed query
f) an internal server error
Currently, a) and b) are returned with HTTP 200 and must do so, because the HTTP status code cannot be changed once the server has started to send payload.
c) is HTTP 504, d) is HTTP 429, e) is HTTP 400, and f) is HTTP 500.

What I would like to see is for each of the cases a)-f):
- what the standards (from W3C, maybe also IETF RFC might matter) allow and/or suggest
- what Apache and Nginx support
- what Firefox and other clients (Wget, curl, maybe Chromium) do
with regard to whether an response payload is possible, preflighting must or can be done, and special headers are possible or required.

I can help with an account on dev.overpass-api.de . An Apache instance is running there, and I would grant you sudo rights if you can install Nginx and get it running in addition to Apache. However, the whole HTTP-status-code-CORS complex competes with general development (some extra commands, better history access, resolving performance bottlenecks), and I deem it less important than these. Hence I won't do this myself.

**Comment by mmd-osm** [comment]
@drolbr : As a short term workaround I could make sense to run the queries in question via command line `osm3s_query` and put the results on the server for @cliffordsnow to download. In that case, we're not restricted by apache timeouts. I do this all the time when doing performance measurements. And yes, this issue has hit me a number of times already, but I don't see an immediate solution to it.

**Comment by cliffordsnow** [comment]
Thanks for the offer, but you do not need to run the query. I will explore pulling the data from a Washington State Planet dump.

**Comment by tyrasd** [comment]
closing as there is no immediate action to be taken (at least on overpass turbo's side of the issue).
